modulemd/v2/modulemd-module-stream-v1.c

```diff
@@ -2353,3 +2353,182 @@ mmd_variant_from_sequence (yaml_parser_t *parser, GError **error)
 
   return g_variant_ref_sink (g_variant_builder_end (&builder));
 }
+
+static gboolean
+emit_yaml_variant (GVariant *variant, yaml_emitter_t *emitter, GError **error)
+{
+  GVariantIter iter;
+  g_autofree gchar *key = NULL;
+  g_autoptr (GVariant) value = NULL;
+
+  if (g_variant_is_of_type (variant, G_VARIANT_TYPE_STRING))
+    {
+      EMIT_KEY (emitter, error, g_variant_get_string (variant, NULL));
+    }
+  else if (g_variant_is_of_type (variant, G_VARIANT_TYPE_BOOLEAN))
+    {
+      if (g_variant_get_boolean (variant))
+        EMIT_KEY (emitter, error, "TRUE");
+      else
+        EMIT_KEY (emitter, error, "FALSE");
+    }
+  else if (g_variant_is_of_type (variant, G_VARIANT_TYPE_DICTIONARY))
+    {
+      EMIT_MAPPING_START (emitter, error);
+      g_variant_iter_init (&iter, variant);
+      while (g_variant_iter_next (&iter, "{sv}", &key, &value))
+        {
+          EMIT_KEY (emitter, error, key);
+          if (!emit_yaml_variant (value, emitter, error))
+            return FALSE;
+          g_clear_pointer (&value, g_variant_unref);
+        }
+      EMIT_MAPPING_END (emitter, error);
+    }
+  else if (g_variant_is_of_type (variant, G_VARIANT_TYPE_ARRAY))
+    {
+      EMIT_SEQUENCE_START (emitter, error);
+      g_variant_iter_init (&iter, variant);
+      while (g_variant_iter_next (&iter, "{v}", &value))
+        {
+          if (!emit_yaml_variant (value, emitter, error))
+            return FALSE;
+          g_clear_pointer (&value, g_variant_unref);
+        }
+      EMIT_SEQUENCE_END (emitter, error);
+    }
+  else
+    {
+      g_set_error (error,
+                   MODULEMD_YAML_ERROR,
+                   MODULEMD_YAML_ERROR_EMIT,
+                   "Unhandled variant type");
+      return FALSE;
+    }
+  return TRUE;
+}
+
+gboolean
+modulemd_module_stream_v1_emit_yaml (ModulemdModuleStreamV1 *self,
+                                     yaml_emitter_t *emitter,
+                                     GError **error)
+{
+  MODULEMD_INIT_TRACE ();
+  if (!modulemd_module_stream_emit_yaml_base (
+        MODULEMD_MODULE_STREAM (self), emitter, error))
+    return FALSE;
+
+  EMIT_SCALAR_IF_SET (
+    emitter, error, "arch", modulemd_module_stream_v1_get_arch (self));
+  EMIT_SCALAR_IF_SET (emitter, error, "summary", self->summary);
+  EMIT_SCALAR_IF_SET (emitter, error, "description", self->description);
+
+  EMIT_HASHTABLE_VALUES (emitter,
+                         error,
+                         "servicelevels",
+                         self->servicelevels,
+                         modulemd_service_level_emit_yaml);
+
+  if (NON_EMPTY_TABLE (self->module_licenses) ||
```

Comment:
module licenses are mandatory for ModuleStreamV1 (and V2, but possibly not a forthcoming V3). If it's not present, this should fail.